pinot-core/src/main/java/org/apache/pinot/core/plan/FilterPlanNode.java

```diff
@@ -122,8 +127,29 @@ private BaseFilterOperator constructPhysicalOperator(FilterContext filter,
           DataSource dataSource = _indexSegment.getDataSource(lhs.getIdentifier());
           switch (predicate.getType()) {
             case TEXT_MATCH:
-              return new TextMatchFilterOperator(dataSource.getTextIndex(),
-                  ((TextMatchPredicate) predicate).getValue(), _numDocs);
+                return new TextMatchFilterOperator(
+                        dataSource.getTextIndex(),
+                        ((TextMatchPredicate) predicate).getValue(),
+                        _numDocs);
+            case REGEXP_LIKE:
+              PredicateEvaluator evaluator = null;
+              if (dataSource.getFSTIndex() != null) {
+                evaluator = FSTBasedRegexpPredicateEvaluatorFactory.newFSTBasedEvaluator(
+                        dataSource.getFSTIndex(),
+                        dataSource.getDictionary(),
+                        ((RegexpLikePredicate) predicate).getValue());
+              } else if (dataSource instanceof MutableDataSource &&
```

Comment:
Why are we checking for the type of data source. getFSTIndex() should return a reader that could have come from a mutable or immutable data source

Reply:
Added comments, also see my response on above comment (from FSTBasedRegexpPredicateEvaluatorFactory.java)